### PR TITLE
Colorful Status Update System

### PR DESCRIPTION
**Pull request by theCydonian**
##### Description

Added a system for adding colorful backgrounds to status updates. This works on both profile and townsquare. Text is displayed on top of an image when a background is selected. Tags are still applied but not displayed for simplicity.

##### Refers/Fixes

Fixes: https://github.com/gitcoinco/web/issues/6194

##### Testing

Test swapping between all combinations attachment types to ensure multiple attachments cannot be selected. Test with maximum message length. Test typical message. Repeat for both townsquare and profile.
https://www.dropbox.com/s/tb3q6qc6rwnuts9/Screen%20Recording%202020-03-19%20at%2017.14.20.mov?dl=0

### CODE REVIEW

**Comment by owocki** [comment]
this is awesome!!! 

CC @octavioamu https://bits.owocki.com/nOuNKA9X

File: app/retail/templates/shared/activity.html [hunk 1]
```diff
@@ -243,63 +245,62 @@
         {% endif %}
       {% endif %}
       </div>
-      <div class="activity-tags activity-tag-first align-items-center">
-        <span class="mx-0">
+        <div class="activity-tags activity-tag-first align-items-center">
+          <span class="mx-0">
```
Comment (thelostone-mc) [inline app/retail/templates/shared/activity.html]: could we de-indent this

File: app/retail/templates/shared/activity.html [hunk 2]
```diff
@@ -243,63 +245,62 @@
...
+              </span>
+            {% endif %}
+
+            {% if not hide_likes %}
+                <a data-pk={{row.pk}} data-username={{row.profile.handle}} class="tip_activity" href="#"
+                data-toggle="tooltip" title="This activity feed has generated {{ row.tip_count_eth|floatformat:2 }} ETH (${{row.tip_count_usd|floatformat:2}}) in Tips."
+                >
+                  <i class="fab fa-ethereum grey" ></i>
+                  <span class="amount grey">{{ row.tip_count_eth|floatformat:3}}</span>
+                </a>
```
Comment (thelostone-mc) [inline app/retail/templates/shared/activity.html]: re-indent by 2 instead of 4

File: app/townsquare/templates/townsquare/shared/shareactivity.html [hunk 3]
```diff
@@ -16,18 +16,21 @@
...
-                      <button id="video-button" class="btn btn-clean position-absolute" data-toggle="tooltip" title="Add a video call to post." style="bottom: 3.5em; right: 1.5em;"><i class="fas fa-video"></i></i></button>
+                      <button id="video-button" class="btn btn-clean position-absolute" data-toggle="tooltip" title="Add a video call to post." style="bottom: 2em; right: 1.5em;"><i class="fas fa-video"></i></i></button>
                     {% endif %}
-                      <button class="btn btn-clean position-absolute click-gif"  data-toggle="tooltip" title="Add a gif to post." data-toggle="dropdown" id="btn_gif" aria-haspopup="true" aria-expanded="false" style="bottom: 2em; right: 1.5em;">
-                          <i class="fas fa-images mr-2"></i>
-                      </button>
-
+                    <button class="btn btn-clean position-absolute click-gif"  data-toggle="tooltip" title="Add a gif to post." data-toggle="dropdown" id="btn_gif" aria-haspopup="true" aria-expanded="false" style="bottom: 3.5em; right: 1.5em;">
+                        <i class="fas fa-images mr-2"></i>
+                    </button>
```
Comment (thelostone-mc) [inline app/townsquare/templates/townsquare/shared/shareactivity.html]: ^ re-indent by 2 instead of 4

**Comment by theCydonian** [comment]
I’ll make the change early tomorrow morning. Atom can auto fix stuff like this.

**Comment by theCydonian** [comment]
The indentation should be good now

**Comment by thelostone-mc** [comment]
@theCydonian code looks alright 🙌 
Could you tell me the need for these files ` signatory/*` & `build/config.gypi ` ?

**Comment by theCydonian** [comment]
@thelostone-mc oh, sorry. Those were accidentally added when I was installing node packages on another project. They have now been removed.

I don't know what to do with the travis-ci failure though. It looks like it fails somewhere I didn't modify.

**Comment by danlipert** [comment]
@theCydonian I fixed the unrelated travis errors, looks like it doesn't like your indentation still. Please fix and then let us know once its ready -  thanks!

**Comment by theCydonian** [comment]
@danlipert I ran eslint fix and then eslint and no errors were given. Should it be fine to submit now?

File: app/assets/v2/css/activity_stream.css [hunk 4]
```diff
@@ -52,6 +52,63 @@
...
+  white-space: nowrap;
+}
+
+.bg-thumbnail {
+  display: inline-block;
+  height: 2.35em;
+  width: 2em;
+  margin-left:0.2em;
+  margin-right: 0.2em;
+  }
```
Comment (danlipert) [inline app/assets/v2/css/activity_stream.css]: messed up indent here

File: app/assets/v2/css/activity_stream.css [hunk 5]
```diff
@@ -52,6 +52,63 @@
   text-align: left;
 }
 
+.hide {
```
Comment (octavioamu) [inline app/assets/v2/css/activity_stream.css:55]: we have many classes for display:none already as we use bootstrap you can use d-none for example.

File: app/dashboard/templates/profiles/status_box.html [hunk 6]
```diff
@@ -33,6 +34,7 @@
       </div>
       <button id="poll-button" class="btn btn-sm p-1" data-toggle="tooltip" title="Add a poll to post." style="bottom: 2em; right: 0em;"><i class="fas fa-fw fa-chart-bar"></i></i></button>
     </div>
+    <div class="hide" id="bg-selector">
```
Comment (octavioamu) [inline app/dashboard/templates/profiles/status_box.html:37]: missing closing tag

File: app/assets/v2/js/status.js [hunk 7]
```diff
@@ -218,22 +267,69 @@ $(document).ready(function() {
...
+  });
+
+  var selectedElement = null;
+  // handle background selection
 
-  // handle poll button
+  $('.bg-thumbnail').click(function(e) {
+    e.preventDefault();
+
+    $('#bg-selector').children('div').children('div').addClass('hide');
```
Comment (octavioamu) [inline app/assets/v2/js/status.js:285]: maybe .find('. selector-bar')  instead?

File: app/assets/v2/js/status.js [hunk 8]
```diff
@@ -363,7 +470,9 @@ $(document).ready(function() {
         data.append('image', image);
       }
     }
-
+    $('#bg-selector').attr('data-selected', null);
+    $('#bg-selector').addClass('hide');
+    $('#bg-selector').children('div').children('div').addClass('hide');
```
Comment (octavioamu) [inline app/assets/v2/js/status.js:475]: find()? or something more readable also can work.

File: app/townsquare/templates/townsquare/shared/shareactivity.html [hunk 9]
```diff
@@ -16,11 +16,13 @@
                       id="textarea"
                       class="form__input border-0 pl-2 status-textarea"
                       placeholder="{% if placeholder %}{{placeholder}}{%else%}What are you thinking?{%endif%}"
+                      style="resize: none; margin-right: 36px; height: 4.5em;"
```
Comment (octavioamu) [inline app/townsquare/templates/townsquare/shared/shareactivity.html:19]: I believe this is now diff on master. Is using a plugin to grow the area

File: package.json [hunk 10]
```diff
@@ -21,7 +21,7 @@
   },
   "homepage": "https://github.com/gitcoinco/web#readme",
   "devDependencies": {
-    "cypress": "^3.4.1",
+    "cypress": "^3.8.3",
```
Comment (octavioamu) [inline package.json:24]: I think this is not related to the PR.

**Comment by octavioamu** [comment]
@theCydonian Seems there are some problems we will need to revert this PR.
- Backgrounds works on kudos but not in townsquare (addition missing on the template)
- Other code was removed (this PR removes favorite button https://github.com/gitcoinco/web/pull/6255/files#diff-3d2c809872e6c16caeddfd561ffef458L301 ) 
Please check deeply this changes and recreate the PR. 
Thanks

**Comment by owocki** [comment]
sorry to see this backed out. can we get those fixed up and put them in the next release?

**Comment by theCydonian** [comment]
I’ll fix as soon as I can

**Comment by theCydonian** [comment]
I readded the favorite button, I think I accidentally deleted when resolving merge conflicts because I had moved that section of the code. I'll look into where I lost the background functionality tomorrow morning.

**Comment by theCydonian** [comment]
It should work as intended now

**Comment by octavioamu** [comment]
@theCydonian  this PR was already merged and then reverted sadly there is not way to turn it into open PR again, to be able to review it and have the latest master you need to create a new PR. 
Could you do that and tag me?

**Comment by octavioamu** [comment]
I created #6532 and did some image optimizations

**Comment by octavioamu** [comment]
@theCydonian don't worry we are going to take it from here. thanks
